packages/ember-data/lib/serializers/rest_serializer.js

```diff
@@ -527,25 +525,43 @@ var RESTSerializer = JSONSerializer.extend({
     ```
 
     In order for Ember Data to know that the model corresponding to
-    the 'response-fast-car' hash is `FastCar` (typeKey: 'fastCar'),
-    you can override typeForRoot to convert 'response-fast-car' to
-    'fastCar' like so:
+    the `response-fast-car` property is the Globals-mode class
+    `App.FastCar` the `typeForRoot` must be overridden. For example:
 
     ```js
     App.ApplicationSerializer = DS.RESTSerializer.extend({
       typeForRoot: function(root) {
         // 'response-fast-car' should become 'fast-car'
         var subRoot = root.substring(9);
+        // normalizes 'fast-car' to 'fastCar'
+        // singularize for good measure
+        return subRoot.singularize().camelize();
+      }
+    });
+    ```
 
-        // _super normalizes 'fast-car' to 'fastCar'
-        return this._super(subRoot);
+    The default implementation of `typeForRoot` is to
+    `singularize` then `camelize` a payload property name.
+    This does not technically match the conventions of
+    Ember-CLI, and although it will work well in apps
+    and acceptance tests, may be problematic in unit tests.
+
+    A suggested implementation for Ember-CLI users would
+    be:
+
+    ```js
+    // app/serializers/application.js
+    import DS from "ember-data";
+    export default DS.RESTSerializer.extend({
+      typeForRoot: function(root) {
+        return subRoot.singularize().dasherize();
```

Comment:
Does this mean Ember CLI users will still need to use a custom Serializer after this pr is merged?